syntax = "proto3";

package nddiwall;

service NddiWall {
  // Initializes the NDDI Display
  rpc Initialize (InitializeRequest) returns (StatusReply) {}
}

// The request for initializing the NDDI Display
message InitializeRequest {
  repeated int32 frameVolumeDimensionalSizes = 1;
  int32 displayWidth = 2;
  int32 displayHeight = 3;
  int32 numCoefficientPlanes = 4;
  int32 inputVectorSize = 5;
}

// Simple status response
message StatusReply {
  enum Status {
    OK = 0;
    NOT_OK = 1;
  }
  Status status = 1;
}
